### PR TITLE
curvebs: add some topo metric

### PR DESCRIPTION
**Pull request by SeanHai**
### What problem does this PR solve?

Issue Number: #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
cicheck

**Comment by wuhongsong** [comment]
@SeanHai continue

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
cicheck

File: src/mds/topology/topology_metric.h [hunk 1]
```diff
@@ -186,6 +188,8 @@ struct LogicalPoolMetric {
     bvar::Status<uint64_t> writeIOPS;
 
     explicit LogicalPoolMetric(const std::string &logicalPoolName) :
+        serverNum(kTopologyLogicalPoolMetricPrefix,
+            logicalPoolName + "_server_num", 0),
         chunkServerNum(kTopologyLogicalPoolMetricPrefix,
             logicalPoolName + "_chunkserver_num", 0),
         copysetNum(kTopologyLogicalPoolMetricPrefix,
```
Comment (Cyber-SiKu) [inline src/mds/topology/topology_metric.h:195]: Why not remove the extra `_ `here?

File: src/mds/topology/topology_metric.cpp [hunk 2]
```diff
@@ -269,27 +271,33 @@ void TopologyMetricService::UpdateTopologyMetrics() {
...
         clusterWriteRate +=  metricIter->second->writeRate.get_value();
         copysetNum += metricIter->second->copysetNum.get_value();
+        serverNum += metricIter->second->serverNum.get_value();
+        logicalCapacity += metricIter->second->logicalCapacity.get_value();
+        logicalAlloc += metricIter->second->logicalAlloc.get_value();
     }
 
-    auto servers = topo_->GetServerInCluster();
-    gClusterMetrics->serverNum.set_value(servers.size());
+    gClusterMetrics->serverNum.set_value(serverNum);
```
[the remark this answers is not in the document]
Comment (SeanHai) [reply src/mds/topology/topology_metric.cpp:291]: > Why change it to this?

The serverNum metric also needed in pool metric, so add it first and the cluster serverNum is sum all pool serverNum metric value, not need to find in topo.

File: src/mds/topology/topology_metric.h (hunk 1 above)
Comment (SeanHai) [reply src/mds/topology/topology_metric.h:195]: > Why not remove the extra `_ `here?

The cluster metric prefix is 'topology_metric_cluster_' which end with '', but pool metric is pool prefix 'topology_metric_logicalPool' + poolName which not end with '-'.

**Review by Cyber-SiKu — APPROVED** [review]

**Review by wuhongsong — APPROVED** [review]

**Pull request merged**